### PR TITLE
[bugfix] Fix handling of symbolic links in `osext.copytree()` when destination exists

### PR DESCRIPTION
**Pull request by teojgo**
* This fixes the inconsistency of handling the symlinks when
  the directories already exist.

* Add a unittest.

Related to #1189

### CODE REVIEW

File: unittests/test_utility.py [hunk 1]
```diff
@@ -406,6 +406,30 @@ def test_virtual_copy_linkparent(direntries):
         osext.copytree_virtual(*direntries, file_links, dirs_exist_ok=True)
 
 
+def test_virtual_copy_symlinks_dirs_exist(tmp_path):
+    src = tmp_path / 'src'
+    src.mkdir()
+    dst = tmp_path / 'dst'
+    dst.mkdir()
+    foo = src / 'foo'
+    foo.write_text('Hello Foo')
```
Comment (vkarak) [inline unittests/test_utility.py:415]: Just touch the file. We don't care about the contents.

File: unittests/test_utility.py [hunk 2]
```diff
@@ -406,6 +406,30 @@ def test_virtual_copy_linkparent(direntries):
...
+    foo_link = src / 'foo.link'
+    foo_link.symlink_to(foo)
+    osext.copytree_virtual(src, dst, symlinks=False, dirs_exist_ok=True)
+    assert (dst / 'foo').exists()
+    assert (dst / 'foo.link').exists()
+    assert (dst / 'foo.link').read_text() == 'Hello Foo'
```
Comment (vkarak) [inline unittests/test_utility.py:421]: Remove this, we don't care.

File: unittests/test_utility.py [hunk 3]
```diff
@@ -406,6 +406,30 @@ def test_virtual_copy_linkparent(direntries):
         osext.copytree_virtual(*direntries, file_links, dirs_exist_ok=True)
 
 
+def test_virtual_copy_symlinks_dirs_exist(tmp_path):
```
Comment (vkarak) [inline unittests/test_utility.py:409]: I suggest parameterizing the test on the `symlinks` argument. Check  how this is done for the `dirs_exist_ok`:

https://github.com/eth-cscs/reframe/blob/3b68ca0c93d227b284163c36f42afe9f2268c806/unittests/test_utility.py#L83-L84

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]
lgtm

**Review by ekouts — APPROVED** [review]

**Pull request merged**